tests/misra/test_mutation.py

```diff
@@ -57,12 +61,19 @@
   r"$a #define auto 1\n",
   # misra-c2012-20.5
   r"$a #define TEST 1\n#undef TEST\n",
+  # unusedFunction
+  r"$a void test(void) {}",
 ]
 
 all_files = glob.glob('board/**', root_dir=ROOT, recursive=True)
 files = [f for f in all_files if f.endswith(('.c', '.h')) and not f.startswith(IGNORED_PATHS)]
 assert len(files) > 70, all(d in files for d in ('board/main.c', 'board/stm32f4/llbxcan.h', 'board/stm32h7/llfdcan.h', 'board/safety/safety_toyota.h'))
 
+# Remove cppcheck files from previous runs to speed tree copying
+for f in glob.glob(f"{ROOT}/**/*", recursive=True):
```

Comment:
I tried amending it, but then started digging more why I am getting all those big dump files created in the ROOT in the first place. I realized test_misra.sh was running before files were even copied to temp file.  Heck, then even when I removed any reference of it from test_mutation.py.  

The issue was the naming. Turns out, Pytest workers were executing anything that start with `test` that is in the same directory - including `test_misra.sh`.  
I renamed test_misra.sh for now to check the CI, but maybe it would be better to ignore it in [pyproject.toml](https://github.com/commaai/panda/commit/78f67d3526f2eb6a634455a4bfd15c17975166d3#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L14 ) ?